Kudu.Core/Settings/DiagnosticsSettingsManager.cs

```diff
@@ -69,6 +69,12 @@ private DiagnosticsSettings ReadSettings()
                 {
                     string fileContent = null;
                     OperationManager.Attempt(() => fileContent = FileSystemHelpers.ReadAllTextFromFile(_path));
+                    if (DiagnosticsSettings.AutoCorrect(_tracer, fileContent, out fileContent))
+                    {
+                        // if there is corrupted value, reset it back to file system
+                        FileSystemHelpers.WriteAllTextToFile(_path, fileContent);
+                    }
+
                     return JsonConvert.DeserializeObject<DiagnosticsSettings>(fileContent);
```

Comment:
We might be over-doing this (given we fix the setter and this would become rare).  I would do the following.  
```CSharp
try
{
    return JsonConvert.DeserializeObject<DiagnosticsSettings>(fileContent);
}
catch (Exception ex)
{
    _tracer.TraceError(ex);
    FileSystemHelpers.SafeDelete(_path);
    return new DiagnosticsSettings();
}
```